test/distributed/_tensor/test_dtensor.py

```diff
@@ -820,6 +820,53 @@ def test_split_tensor_1D(self) -> None:
                 ]
                 assert_array_equal(expected_is_tensor_empty, is_tensor_empty)
 
+    @with_comms
+    def test_weight_tying(self):
+        class TestModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                # Initialize different weights for embedding and fc.
+                torch.manual_seed(1)
+                self.embedding = torch.nn.Embedding(16, 8)
+                torch.manual_seed(2)
```

Comment:
two questions:
1. is there any reason why we need to manually set seed twice? If we want to control the randomness, we only need to have the first `manual_seed`.
2. why do we have it inside the class constructor `__init__`? This means, every `TestModule` object will have the same embedding and Linear parameter. Is that what we want? If we only want to control the randomness so that every time the test runs the initialization result will be the same, we can put `manual_seed` before the line `model = TestModule().to(self.device_type)` (actually this is even unnecessary because [`MultiProcessTestCase`](https://github.com/pytorch/pytorch/blob/6243dbb5c08e9b3f87bc7dc74816b2ceb6957a94/torch/testing/_internal/common_distributed.py#L961) will call `manual_seed(0)` and `DTensorTestBase` is a subclass of `MultiProcessTestCase`).

Reply:
So my point was to make sure two submodules have different weights. Not setting manual seeds twice should also do the work (just by randomness), but mentally it doesn't tell people the two initializations should be different. An alternative I had in mind is to do manual initialization rather than relying on internal gaussian distribution. Maybe I should do that?

Reply:
I think this is just to test "weight tying" and a way to make sure embedding/linear have different weights initialized. I am fine with either approach.